erpnext/hr/doctype/leave_type/leave_type.py

```diff
@@ -5,13 +5,20 @@
 import calendar
 import frappe
 from datetime import datetime
+from frappe.utils import today
 from frappe import _
 
 from frappe.model.document import Document
 
 class LeaveType(Document):
 	def validate(self):
 		if self.is_lwp:
-			leave_allocation = frappe.db.sql_list("""select name from `tabLeave Allocation` where leave_type=%s""", (self.name))
+			leave_allocation = frappe.get_all("Leave Allocation", filters={
+				'leave_type': self.name,
+				'from_date': ("<=", today()),
+				'to_date': (">=", today())
+			}, ['name'])
+			leave_allocation = [l['name'] for l in leave_allocation]
+			frappe.db("""select name from `tabLeave Allocation` where leave_type=%s""", (self.name))
```

Comment:
??

Reply:
It checks if the user is trying to change leave type when a leave allocation is active. If the leave allocation is not currently active, the user can change the leave type.